tokio-stream/tests/mpsc_bounded_stream.rs

```diff
@@ -0,0 +1,71 @@
+use futures::{Stream, StreamExt};
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::ReceiverStream;
+
+#[tokio::test]
+async fn size_hint_stream_open() {
+    let (tx, rx) = mpsc::channel(4);
+
+    tx.send(1).await.unwrap();
+    tx.send(2).await.unwrap();
+
+    let mut stream = ReceiverStream::new(rx);
+
+    assert_eq!(stream.size_hint(), (2, None));
+    stream.next().await;
+    assert_eq!(stream.size_hint(), (1, None));
+    stream.next().await;
+    assert_eq!(stream.size_hint(), (0, None));
+}
+
+#[tokio::test]
+async fn size_hint_stream_closed() {
+    let (tx, rx) = mpsc::channel(4);
+
+    tx.send(1).await.unwrap();
+    tx.send(2).await.unwrap();
+
+    let mut stream = ReceiverStream::new(rx);
+    stream.close();
```

Comment:
A stream may be closed both by calling `Receiver::close` and by dropping all senders. These use different mechanisms internally, so it would be good to test both scenarios.